src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/FormattingPassBase.cs

```diff
@@ -455,6 +455,11 @@ private void CleanupSourceMappingStart(FormattingContext context, Range sourceMa
 
             // At this point, `contentIndentLevel` should contain the correct indentation level for `}` in the above example.
             var replacement = context.NewLineString + context.GetIndentationLevelString(contentIndentLevel);
+            for (var i = 0; i < newLineCount - 1; i++)
+            {
+                // Make sure to preserve the same number of blank lines as the original string had
+                replacement = context.NewLineString + replacement;
+            }
```

Comment:
any worry about allocations here?  or is the expectation that this is always a tiny set so gc will clean it up?

consider a stringbuilder just to be on the safe side?

Reply:
While it's not common for code to contain large numbers of blank lines, I'll switch to `StringBuilder` to avoid the O(n²) algorithm.